README.md

```diff
@@ -13,3 +13,113 @@ This application has unit tests, you can run it with command:
+- **Description**: Maximum event loop delay.
+- **Metric**: `nodejs_eventloop_lag_max_seconds`
+- **Condition**: `nodejs_eventloop_lag_max_seconds` > 0.5s.
+
+### Long GC Duration Alert
+- **Source**: Prometheus metrics
+- **Description**: Garbage collection duration.
+- **Metric**: `nodejs_gc_duration_seconds_sum`
+- **Condition**: `nodejs_gc_duration_seconds_sum` > 0.1s in 1m.
+
+### Mail scheduling failed
+- **Source**: Custom metrics
+- **Description**: Cron job didn't work in scheduled time.
+- **Metric**: `notification_mailer_cron`
+- **Condition**: notification_mailer_cron didn't change for 25 hours
+
+### Exchange Rate can't reach rate's sources
+- **Source**: Custom metrics
+- **Description**: Exchange Rate failed too much
+- **Metric**: `all_rate_client_failed`
+- **Condition**: all_rate_client_failed > 5 in 1m.
+
+### Mail sent failed
+- **Source**: Log
+- **Description**: Error happened in email sending
+
+### Infrastructure failing
+- **Source**: Log
+- **Description**: Connection to DB or Kafka failed
+
+## Custom Metrics Documentation
+
+### `rate_update_cron`
+- **Description**: Counter for rate update cron executions
+- **Labels**: None
+
+### `rate_update_notification`
+- **Description**: Counter for rate update notifications
+- **Labels**: None
+
+### `all_rate_client_failed`
+- **Description**: Counter for all failed client requests
+- **Labels**: None
+
+### `exchange_rate_fetched`
+- **Description**: Counter for fetched exchange rates
+- **Labels**: `source`, `status`
+
+### `http_request`
+- **Description**: Counter for HTTP requests to microservice
+- **Labels**: `method`, `path`, `statusCode`
+
+### `subscription_removal_ended_transactions`
+- **Description**: Counter for ended subscription removal transactions
+- **Labels**: `status`
+
+### `kafka_event`
+- **Description**: Counter for Kafka events sent to microservice
+- **Labels**: `eventType`
+
+### `subscription_created`
+- **Description**: Counter for created subscriptions
+- **Labels**: `status`
+
+### `subscription_removal`
+- **Description**: Counter for removed subscriptions
+- **Labels**: `status`
+
+### `customers_created`
+- **Description**: Counter for created customers
+- **Labels**: `status`
+
+### `customers_removal`
+- **Description**: Counter for removed customers
+- **Labels**: `status`
+
+### `rate_requests`
+- **Description**: Counter for rate requests
+- **Labels**: `status`
+
+### `rate_updates`
+- **Description**: Counter for rate updates
+- **Labels**: `status`
+
+### `mailer_subscription_creation`
+- **Description**: Counter for created mailer subscriptions
+- **Labels**: `status`
+
+### `mailer_subscription_removal`
+- **Description**: Counter for removed mailer subscriptions
+- **Labels**: `status`
+
+### `exchange_rate_mail_notification`
+- **Description**: Counter for exchange rate mail notifications
+- **Labels**: `status`
+
+### `notification_mailer_cron`
+- **Description**: Counter for notification mail cron jobs
+- **Labels**: None
```

Comment:
Thank you for your extensive README file. That would be really interesting to know **why** you have chosen these particular metrics to follow. What idea is behind each of the metric/log?)

Reply:
About metrics:
I have 2 types of metrics:

1. **Infrastructure Metrics**: Most of these are generated automatically by prom-client (CPU, memory, event loop lags, etc.), along with some custom metrics that I have created (_http_request_, _kafka_event_). The purpose of these metrics is to measure the global state of the application; they are not related to any specific business logic of the application.
2. **Business Metrics**: Each metric is related to a specific part of the business matter within the application. Examples of business metrics include _customers_created_, _rate_updates_, and _all_rate_client_failed_. The purpose of these metrics is to monitor aspects critical to the product. Some business metrics are also used for alert conditions, for example, an alert (Exchange Rate can't reach rate's sources) with the condition (_all_rate_client_failed_ > 5 in 1 min).

About logs:
There is nothing super specual. Here 4 type of logs:

1. **info**: Provides information pertinent to the business matter regarding the state at that point.
2. **warn**: Indicates potential issues that may not have immediately disrupted operations
3. **error**: Reports errors that have caused a failure in some business-critical operation or infrastructure layer(for example lost DB connection)
4. **debug**: Offers debugging information about the application (e.g., DB, Kafka connections, server start-up, etc.)

Some of error can be very important, so it can trigger alert.

Reply:
Thank you for the extensive explanation :)

I have one more question left about the logs. What did you base your decision on to log error or info or debug?)

Reply:
It depends) If they are related to business logic, they should be categorized as info logs. If an exception occurs, they should be logged as errors. For infrastructure layers like Kafka, databases, HTTP routers, etc., use debug logs